lib/jekyll/converters/scss.rb

```diff
@@ -179,17 +186,52 @@ def sass_configs
         )
       end
 
+      def sass_embedded_config(data)
+        {
+          :data                => data,
+          :file                => file,
+          :indented_syntax     => syntax == :sass,
+          :include_paths       => sass_load_paths,
+          :output_style        => sass_style,
+          :source_map          => sourcemap_required?,
+          :out_file            => output_path,
+          :omit_source_map_url => !sourcemap_required?,
+          :source_map_contents => true,
+        }
+      end
+
       def convert(content)
+        case sass_implementation
+        when :"sass-embedded"
+          require "sass"
+          sass_embedded_convert(content)
+        when :sassc
+          sass_convert(content)
+        end
+      end
+
+      def sass_convert(content)
```

Comment:
Let `:sass_convert` and `:sass_embedded_convert` be private methods.
`:convert` being public is sufficient.